src/averageAges.js

```diff
@@ -15,11 +15,24 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = people.filter(person => {
+    const { sex, died } = person;
+
+    return century
+      ? sex === 'm' && Math.ceil(died / 100) === century
+      : sex === 'm';
+  });
+
+  // * I'm checking if array have some data about people
+  // * (if we input in parameter `century` value which wasn't found)
+  // * I know, I should avoid if (), but that condition do my code more flexible
+  if (filteredMen.length === 0) {
+    return null;
+  }
```

Comment:
You can handle it in `getAverageAge` function which will return 0 if `filteredMen.length === 0`.